test/Main.purs

```diff
@@ -1,9 +1,233 @@
 module Test.Main where
 
 import Prelude
+
 import Control.Monad.Eff (Eff)
-import Control.Monad.Eff.Console (CONSOLE, log)
+import Control.Monad.Eff.AVar (AVAR)
+import Control.Monad.Eff.Console (CONSOLE)
+import Data.Array (sort)
+import Data.Either (Either(..))
+import Data.Fuzzy (Fuzzy(..), FuzzyStr(..), Rank(..), match, matchStr)
+import Data.Generic.Rep (class Generic)
+import Data.Generic.Rep.Eq (genericEq)
+import Data.Generic.Rep.Show (genericShow)
+import Data.Maybe (Maybe(..))
+import Data.StrMap (StrMap, fromFoldable)
+import Data.Tuple (Tuple(..))
+import Test.Unit (suite, test)
+import Test.Unit.Assert (equal)
+import Test.Unit.Console (TESTOUTPUT)
+import Test.Unit.Main (runTest)
+
+newtype TestRecord = TR { name :: String, value :: String }
+derive instance genericTestRecord :: Generic TestRecord _
+instance eqTestRecord :: Eq TestRecord where eq = genericEq
+instance showTestRecord :: Show TestRecord where show = genericShow
+
```

Comment:
This could be a good place to check instances of classes with explicit laws like `Semiring` using property-based testing like this:

https://pursuit.purescript.org/packages/purescript-strongcheck/3.1.0/docs/Test.StrongCheck

Reply:
or https://github.com/garyb/purescript-quickcheck-laws ;)